desktop/core/src/desktop/js/reactComponents/FileChooser/FileChooserModal/FileChooserModal.tsx

```diff
@@ -11,23 +34,53 @@ interface FileProps {
 const defaultProps = { title: 'Choose a file', okText: 'Select' };
 
 const FileChooserModal: React.FC<FileProps> = ({ show, onCancel, title, okText }) => {
+  const [fileSystemList, setFileSystemList] = useState<Array<FileSystem>>([]);
+
+  const icons = {
+    hdfs: <HdfsIcon />,
+    abfs: <AdlsIcon />,
+    s3: <S3Icon />
+  };
+
   const handleOk = () => {
-    //temporary until the file is selected through the file chooser compoent
+    //temporary until the file is selected through the file chooser component
     onCancel();
   };
 
   const handleCancel = () => {
     onCancel();
   };
 
+  useEffect(() => {
+    if (show) {
```

Comment:
If the user shows the modal (for the first time) and clicks the menu FileSystemList before this api call has returned the list will be empty and there is no indication that something is getting loaded. We should address that somehow, perhaps with a loading indicator in the modal body. 

Is there a good reason we call the fetchFileSystems each time the Modal is shown? If not we can just fetch it once when the component is created, still using `useEffect` but with a check if the `fileSystemList` state is undefined (so then initial state can't be an empty array). 

Another option is to have the consumer of this component be responsible for passing in the fileSystemsObj using a prop. If it turns out that fetchFileSystems returns an empty list then we probably shouldn't give the user the option to show the FileChooserModal to begin with. Just thinking out loud here :-)